<launch>
    <arg name="is_sim" default="0"/>
    <arg name="is_physical" default="1"/>
    <arg name="use_joystick" default="1"/>
    <arg name="use_keyboard" default="0"/>
    <arg name="serial_port" default="/dev/ttyS0"/>
    <arg name="use_imu" default="0"/>

    <group if="$(arg is_physical)">
        <!-- Communication Node between Arduino and Pi -->
        <!--<node pkg="rosserial_python" type="serial_node.py"
        name="dingo_rosserial" args="$(arg serial_port)" output="screen"/>-->

        <!--<node pkg = "dingo_nano_interfacing" type = "battery_voltage_checking.py" name = "battery_voltage" output="screen"></node>-->

        <!-- LCD Screen -->
        <node pkg = "dingo_peripheral_interfacing" type = "dingo_lcd_interfacing.py" name = "dingo_LCD_node" output="screen"></node>
    </group>

    <!-- If want to use the joystick, launch joystick node -->
    <group if="$(arg use_joystick)">
        <node pkg="joy" type="joy_node" name="JOYSTICK">
            <param name="autorepeat_rate" value="30"/>
            <!-- <param name="joy_node/dev" value="/dev/input/js0"/>
            <arg name="coalesce_interval" value="0.02"/>
            -->
        </node>
    </group>

    <group if="$(arg use_keyboard)">
        <node pkg = "dingo_input_interfacing" type = "Keyboard.py" name = "keyboard_input_listener" output="screen"></node>
    </group>
    
    <!-- Dingo Driver -->
    <node pkg="dingo" type="dingo_driver.py"
        name="dingo" args="$(arg is_sim) $(arg is_physical) $(arg use_imu)" output="screen"/>
</launch>
    
